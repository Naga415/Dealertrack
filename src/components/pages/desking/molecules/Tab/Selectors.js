'use strict';
import React from 'react';
import RadioButtonList from '@coxautokc/fusion-ui-components/lib/RadioButtonList';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class Selectors extends React.PureComponent {
  shouldComponentUpdate(nextProps){
    return this.props.selectedValue!== nextProps.selectedValue;
  }

  render() {
    const selectors = [];
    React.Children.forEach(this.props.children, function (element) {
      selectors.push({
        value: element.props.label,
        label: element.props.label
      });
    }, this);

    return (
      <RadioButtonList
        className="primary"
        displayLabel={false}
        inline
        required
        htmlId="DealType"
        name="DealType"
        label="DealType"
        onChange={this.props.onChange}
        value={this.props.selectedValue}
        options={selectors}
      />
    );
  }
}
Selectors.propTypes = {
  children: PropTypes.node,
  onChange: PropTypes.func.isRequired,
  selectedValue: PropTypes.string.isRequired
};
export default Selectors;
