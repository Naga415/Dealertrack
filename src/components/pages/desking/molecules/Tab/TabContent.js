'use strict';
import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class TabContent extends React.PureComponent {
  render() {
    return (
      <div id={this.props.htmlId} className='tab__container__content'>
        {this.props.children}
      </div>
    );
  }
}
TabContent.defaultProps = {
  isDefault: false
};

TabContent.propTypes = {
  children: PropTypes.node,
  htmlId: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default TabContent;
