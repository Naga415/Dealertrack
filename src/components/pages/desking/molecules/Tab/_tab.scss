$tab:'.tab'; // namespace for component

#{$tab} {
  padding: 10px 0;

  &__selectors {
    padding: 0;

    .selectInput {
      padding-right: 15px;
    }
  }

  &__container {
    padding: 5px;

    &__content {
      padding: 0;
    }
  }
}
