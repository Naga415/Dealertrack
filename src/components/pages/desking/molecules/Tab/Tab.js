'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import TabContent from './TabContent';
import Selectors from './Selectors';

// eslint-disable-next-line react/prefer-stateless-function
class Tab extends React.PureComponent {
  constructor(props, context) {
    super(props, context);
  }

  setCurrentTab = () => {
    const nodes = React.Children.toArray(this.props.children);
    const selectedValue = this.props.selectedValue;
    const currentTab = nodes.find(function (element) {
      if (element.type === TabContent && element.props.label === selectedValue)
        return element;
    });
    return currentTab;
  };

  onChange = (event) => {
    this.props.onTabSelect(event.target.value);
    this.setCurrentTab();
  };

  buildSelectors = () => {

  };

  render() {
    return (
      <div className="tab" id={this.props.htmlId}>
        <div className="tab__selectors">
          <Selectors onChange={this.onChange} selectedValue={this.props.selectedValue} children={this.props.children}/>
        </div>
        <div className="tab__container">
          {this.setCurrentTab()}
        </div>
      </div>
    );
  }
}

Tab.propTypes = {
  children: PropTypes.node,
  htmlId: PropTypes.string.isRequired,
  onTabSelect: PropTypes.func.isRequired,
  selectedValue: PropTypes.string
};

export default Tab;
