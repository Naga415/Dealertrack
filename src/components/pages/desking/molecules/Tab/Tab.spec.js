import React from 'react';
import renderer from 'react-test-renderer';
import Tab from './Tab';
import TabContent from './TabContent';
import { mount } from 'enzyme';
import Radio from 'react-bootstrap/lib/Radio';

describe('Tab', () => {
  function getTabNode() {
    const onTabSelect = () => {

    };

    const props = {
      htmlId: "Tab",
      selectedValue: "Finance",
      onTabSelect
    };

    return (
      <Tab {...props}>
        <TabContent htmlId="FinanceDeal" label="Finance">
          Finance Content
        </TabContent>
        <TabContent htmlId="LeaseDeal" label="Lease">
          Lease Content
        </TabContent>
        <TabContent htmlId="CashDeal" isDefault={true} label="Cash">
          Cash Content
        </TabContent>
        <TabContent htmlId="BalloonDeal" label="Balloon">
          Balloon Content
        </TabContent>
      </Tab>
    );
  }

  function setupmount() {
    return mount(getTabNode());
  }

  it('renders <Tab/>', () => {
    const component = renderer.create(getTabNode());
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders 4 tab selectors', () => {
    const component = setupmount();
    expect(component.find('.selectInput').length).toBe(4);
  });

  it('renders Tab with the active <TabContent/>', () => {
    const component = setupmount();
    expect(component.find(TabContent).length).toBe(1);
  });

  it('renders Tab with the active <TabContent/> which is Finance', () => {
    const component = setupmount();
    expect(component.find(TabContent).length).toBe(1);
    expect(component.find(TabContent).props().label).toEqual('Finance');
  });

  it('renders Tab with the new active <TabContent/> that corresponds to the clicked selector', () => {
    const component = setupmount();
    const selectors = component.find(Radio);
    expect(selectors.length).toBe(4);
    selectors.find({ value: 'Finance' }).simulate('change');
    expect(component.find(TabContent).props().label).toEqual('Finance');
  });
});
