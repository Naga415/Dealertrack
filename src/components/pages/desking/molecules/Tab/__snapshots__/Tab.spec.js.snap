// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders <Tab/> 1`] = `
<div
  className="tab"
  id="Tab"
>
  <div
    className="tab__selectors"
  >
    <div
      className="primary form-group"
    >
      <label
        className="selectInput radio-inline"
        style={undefined}
        title=""
      >
        <input
          checked={true}
          disabled={false}
          id="DealType"
          name="DealType"
          onChange={[Function]}
          type="radio"
          value="Finance"
        />
        Finance
      </label>
      <label
        className="selectInput radio-inline"
        style={undefined}
        title=""
      >
        <input
          checked={false}
          disabled={false}
          id="DealType"
          name="DealType"
          onChange={[Function]}
          type="radio"
          value="Lease"
        />
        Lease
      </label>
      <label
        className="selectInput radio-inline"
        style={undefined}
        title=""
      >
        <input
          checked={false}
          disabled={false}
          id="DealType"
          name="DealType"
          onChange={[Function]}
          type="radio"
          value="Cash"
        />
        Cash
      </label>
      <label
        className="selectInput radio-inline"
        style={undefined}
        title=""
      >
        <input
          checked={false}
          disabled={false}
          id="DealType"
          name="DealType"
          onChange={[Function]}
          type="radio"
          value="Balloon"
        />
        Balloon
      </label>
      
    </div>
  </div>
  <div
    className="tab__container"
  >
    <div
      className="tab__container__content"
      id="FinanceDeal"
    >
      Finance Content
    </div>
  </div>
</div>
`;
